server/routes/api/integrations/integrations.ts

```diff
@@ -129,9 +141,32 @@ router.post(
     const { user } = ctx.state.auth;
     const { transaction } = ctx.state;
 
-    const integration = await Integration.findByPk(id, { transaction });
+    const integration = await Integration.findByPk(id, {
+      rejectOnEmpty: true,
+      transaction,
+    });
     authorize(user, "delete", integration);
 
+    // If it's GitHub, trigger its API for deletion of this installation
+    if (integration.service === IntegrationService.GitHub) {
```

Comment:
Hmm, this sucks inline – it should be in an async processor, but in order to do that you need the details of the deleted integration, which means it should be soft-deleted or can add the details on the `integrations.delete` event. Thoughts?

Reply:
Async would be better, yea. Question is whether we should delete the record _after_ uninstall is successful from GitHub's end.

Reply:
I would delete our record immediately so it can reflect in the UI and then async delete github (the background job will retry). If it eventually fails for some reason it's not the end of the world as we deleted the auth on our side, it's just cleanup

Reply:
Maybe, we can soft-delete it inside `integrations.delete` and then permanently delete in the async processor after uninstall is successful.

Reply:
No soft delete on this table currently 🙈 . Does also leave the question of how do all the other integrations get hard deleted

Reply:
Yeah, I can add it, no issues.

Reply:
Note that the slack ones will accumulate then since no async delete for them 🙈

Reply:
The processor can just hard delete anything that isn't github 👍